rules/__tests__/valid-expect.test.js

```diff
@@ -131,5 +139,30 @@ ruleTester.run('valid-expect', rules['valid-expect'], {
         },
       ],
     },
+    {
+      code:
+        'test("foo", () => { expect(Promise.resolve(2)).resolves.toBeDefined(); });',
+      errors: [{ message: "Must return or await 'expect.resolves' statement" }],
+    },
+    {
+      code:
+        'test("foo", async () => { expect(Promise.resolve(2)).resolves.toBeDefined(); });',
+      errors: [{ message: "Must await 'expect.resolves' statement" }],
+    },
+    {
+      code:
+        'test("foo", () => { expect(Promise.reject(2)).rejects.toBeDefined(); });',
+      errors: [{ message: "Must return or await 'expect.rejects' statement" }],
+    },
+    {
+      code:
+        'test("foo", async () => { expect(Promise.reject(2)).rejects.toBeDefined(); });',
+      errors: [{ message: "Must await 'expect.rejects' statement" }],
+    },
+    {
+      code:
+        'test("foo", async () => expect(Promise.reject(2)).rejects.toBeDefined());',
```

Comment:
This is technically valid, since this arrow function implicitly returns the `expect()` statement. However, the `test()` callback function is async, so I figured we could warn when you're using `async` without `await` in this case.

Thoughts on this?

Reply:
I think that should be fine as a separate rule.

https://github.com/avajs/eslint-plugin-ava/blob/master/docs/rules/no-async-fn-without-await.md

Reply:
Ok, I can remove that check from this rule then, and we can open an issue for another rule.

Reply:
Created #81 so this wouldn't get lost